speclang_modern.tex

```diff
@@ -2099,6 +2099,40 @@ \subsection{Non-terminating functions}
 to always terminate, which is the expected behavior of most
 functions.
 
+The \lstinline|terminates| clause can also be specified in a statement contract
+with the same meaning. When such a clause is not specified, it inherits the
+one associated to the innermost block containing it (recursively, until a
```

Comment:
Should it really inherit the clause?

```c
/*@ terminates c > 0; */
void f(int c) {
   if (c > 0) { return; }
  else {
     c = 1; // just to confuse the reader
    /*@ ensures \false; */
    while (1);
  }
}
```

It doesn't make sense to say that the statement contract inherits the `terminates c > 0;` clause from the function contract. At best, it can inherit `terminates \at(c>0,Pre);` (which amounts to `\false` in this branch).

Reply:
> At best, it can inherit terminates \at(c>0,Pre); (which amounts to \false in this branch).

This is what I had in mind, but indeed "inherits" may be unclear :thinking:

Reply:
I have tried something